### ISSUE TITLE
[RPC] Guess the language based on the given file

### ISSUE THREAD

**Issue by tkotosz** [phpactor/phpactor#1031]
Currently it seems a `language` parameter need to be passed over to some commands to make it work with non-php files (e.g. gherkin feature files) otherwise phpactor seem to default to `php` language. It would be good to implement a file type guesser in phpactor which could guess the language based on the file. (it could be a very simple one for now based on the file extension ".php" or ".feature")

This way the RPC client would not have to pass over the language parameter and also it would automatically prevent problems when phpactor tries to do something in a file which is not supported (e.g. random ruby or bash script file).

See also https://github.com/phpactor/phpactor/issues/1030

**Comment by dantleech** [comment]
It's currently the clients responsiblity to pass the language - can Sublime do that?

**Comment by tkotosz** [comment]
yep, I can get the "syntax" of the file which is something like `Packages/PHP/PHP.sublime-syntax`, then I can grab the last part and make it lowercase to result in `php` :joy: . But in the long run I think it would be probably better if phpactor could guess the file type. (probably would be useful for LSP as well :thinking: )

**Comment by tkotosz** [comment]
based on the VIM plugin I need to pass over language for these: goto_definition, goto_implementation, goto_type

**Comment by dantleech** [comment]
f.e. the languageID for Behat files is `cucumber`, so `.feature` doesn't help here.

**Comment by tkotosz** [comment]
I went with the easy approach and I just figure out the lang id from the current syntax configured for the file. https://github.com/tkotosz/sublime-phpactor-plugin/commit/fc00032bdda6c8eeda19658d69aadc83a569a3c4
